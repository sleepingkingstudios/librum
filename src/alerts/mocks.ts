import type {
  Alert,
  DismissAlert,
  DismissAllAlerts,
  DisplayAlert,
} from '@alerts';

type MockAlertsContext = {
  alerts: Alert[],
  dismissAlert: jest.MockedFunction<DismissAlert>,
  dismissAllAlerts: jest.MockedFunction<DismissAllAlerts>,
  displayAlert: jest.MockedFunction<DisplayAlert>,
};

type MockUseAlerts = jest.MockedFunction<() => MockAlertsContext>;

const alerts: MockAlertsContext = {
  alerts: [] as Alert[],
  dismissAlert: jest.fn(),
  dismissAllAlerts: jest.fn(),
  displayAlert: jest.fn(),
};

export const useAlerts: MockUseAlerts = jest.fn(() => alerts);
