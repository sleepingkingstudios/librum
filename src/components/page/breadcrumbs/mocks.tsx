import * as React from 'react';

import type { Breadcrumb } from './types';

const formatBreadcrumb = (breadcrumb: Breadcrumb): string => {
  const { active, label, url } = breadcrumb;

  return `${label}${url ? ` @ ${url}` : ''}${active ? ' (active)' : ''}`;
};

export const PageBreadcrumbs = ({
  breadcrumbs,
}: {
  breadcrumbs: Breadcrumb[],
}): JSX.Element => (
  <ul aria-label="breadcrumbs">
    {
      breadcrumbs.map((breadcrumb: Breadcrumb) => (
        <li aria-label="breadcrumb" key={breadcrumb.label}>
          { formatBreadcrumb(breadcrumb) }
        </li>
      ))
    }
  </ul>
);
