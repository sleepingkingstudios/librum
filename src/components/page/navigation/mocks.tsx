import * as React from 'react';

import type { DropdownProps } from './dropdown';
import type { NavigationItemProps } from './item';

const renderNavigationItem = (
  item: DropdownProps | NavigationItemProps
): JSX.Element => {
  const { label } = item;

  if ('items' in item) {
    return (
      <li key={label}>
        { label }
        <ul>
          { item.items.map(renderNavigationItem) }
        </ul>
      </li>
    );
  }

  const { url } = item;

  return (
    <li aria-label="navigation-item" key={label}>{ `${label} @ ${url}` }</li>
  );
};

export const PageNavigation = ({
  navigation,
}: {
  navigation: Array<DropdownProps | NavigationItemProps>,
}): JSX.Element => {
  if (!navigation) { return null; }

  if (navigation.length === 0) { return null; }

  return (
    <ul aria-label="navigation">
      {
        navigation.map(renderNavigationItem)
      }
    </ul>
  );
};
