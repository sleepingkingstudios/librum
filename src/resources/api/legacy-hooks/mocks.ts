import { defaultResponse } from '@api/test-helpers';
import type { UseResourceQuery } from '../types';

const resourceQuery: jest.MockedFunction<UseResourceQuery> =
  jest.fn(() => defaultResponse);

export const useResourceQuery: jest.MockedFunction<() => UseResourceQuery> =
  jest.fn(() => resourceQuery);
