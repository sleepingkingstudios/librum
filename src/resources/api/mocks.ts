import type { UseQuery } from '@api';
import { defaultResult } from '@api/test-helpers';
import type {
  ResourceApiEndpointConfiguration,
  ResourceConfiguration,
} from '../types';
import type { ResourceApiHooks } from './types';

const useIndexResources: jest.MockedFunction<UseQuery> =
  jest.fn(() => defaultResult);

type GenerateResourcesApiProps = {
  endpoints?: ResourceApiEndpointConfiguration,
} & ResourceConfiguration;

type GenerateResourcesApi =
  (options: GenerateResourcesApiProps) => ResourceApiHooks;

export const generateResourcesApi: jest.MockedFunction<GenerateResourcesApi> =
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  jest.fn((...args) => ({
    useIndexResources,
  }));
