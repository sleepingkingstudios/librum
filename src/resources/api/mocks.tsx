import { responseWithStatus } from '@api';
import type {
  Refetch,
  Response,
} from '@api';

type UseResourceQuery = (props: unknown) => [Response, Refetch];

export { generateAlerts } from './utils';

export const useResourceQuery: jest.MockedFunction<UseResourceQuery> = jest.fn(
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  (props) => [responseWithStatus({ status: 'uninitialized' }), jest.fn()]
);
