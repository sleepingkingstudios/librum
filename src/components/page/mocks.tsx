import * as React from 'react';

import type {
  Breadcrumb,
  NavigationProps,
} from './index';
import { isElement } from '@utils/react-utils';
import { PageBreadcrumbs } from './breadcrumbs/mocks';
import { PageNavigation } from './navigation/mocks';

const renderBreadcrumbs = ({
  breadcrumbs,
}: {
  breadcrumbs?: JSX.Element | Breadcrumb[],
}): JSX.Element => {
  if (!breadcrumbs) { return null; }

  if (isElement(breadcrumbs)) { return breadcrumbs; }

  const configured = breadcrumbs as Breadcrumb[];

  if (configured.length === 0) { return null; }

  return (<PageBreadcrumbs breadcrumbs={configured} />);
};

const renderNavigation = ({
  navigation,
}: {
  navigation: JSX.Element | NavigationProps,
}): JSX.Element => {
  if (!navigation) { return null; }

  if (!('length' in navigation)) { return navigation; }

  if (navigation.length === 0) { return null; }

  return (<PageNavigation navigation={navigation} />);
};

const renderTitle = ({
  subtitle,
  title,
}: {
  subtitle?: string,
  title?: string,
}): JSX.Element => {
  if (!title || title.length === 0) { return null; }

  if (!subtitle || subtitle.length === 0) {
    return (<span aria-label="title">{ title }</span>);
  }

  return (<span aria-label="title">{ `${title}: ${subtitle}` }</span>);
};

export const Page = ({
  breadcrumbs,
  children,
  navigation,
  subtitle,
  title,
}: {
  breadcrumbs?: JSX.Element | Breadcrumb[],
  children: React.ReactNode,
  navigation?: JSX.Element | NavigationProps,
  subtitle?: string,
  title?: string,
}): JSX.Element => {
  return (
    <div className="mock-page">
      { renderTitle({ subtitle, title }) }

      { renderNavigation({ navigation }) }

      { children }

      { renderBreadcrumbs({ breadcrumbs }) }
    </div>
  );
};
