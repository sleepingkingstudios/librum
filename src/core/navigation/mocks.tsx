import * as React from 'react';

import { PageNavigation } from '@components/page/navigation/mocks';

const navigation = [
  {
    label: 'Nav Item',
    url: '/nav-item',
  },
  {
    label: 'Dropdown',
    items: [
      {
        label: 'Dropdown Item',
        url: '/dropdown-item',
      },
    ],
  },
];

export const CoreNavigation =
  (): JSX.Element => (<PageNavigation navigation={navigation} />);
