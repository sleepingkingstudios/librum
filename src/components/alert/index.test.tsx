import * as React from 'react';
import { v4 as generateUuid } from 'uuid';
import { faSatelliteDish } from '@fortawesome/free-solid-svg-icons';

import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { render } from '@test-helpers/rendering';

import { Alert } from './index';
import { useAlerts as mockUseAlerts } from '@alerts/mocks';
import type { Alert as IAlert } from '@alerts';

// eslint-disable-next-line @typescript-eslint/no-unsafe-return
jest.mock('@alerts', () => require('@alerts/mocks'));

const { dismissAlert } = mockUseAlerts();

describe('<Alert />', () => {
  const alert: IAlert = {
    dismissable: false,
    message: 'Attempting to connect to Moon base...',
    persistent: false,
    type: 'info',
    uuid: generateUuid(),
  };

  beforeEach(() => { dismissAlert.mockClear(); });

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <Alert alert={alert} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('when the alert is dismissable', () => {
    const dismissableAlert: IAlert = {
      ...alert,
      dismissable: true,
    }

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Alert alert={dismissableAlert} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    describe('dismissing the alert', () => {
      it('should dismiss the alert', async () => {
        const { uuid } = dismissableAlert;
        const { getByRole } = render(
          <Alert alert={dismissableAlert} />
        );

        const button = getByRole('button');

        expect(button).toBeVisible();

        await userEvent.click(button);

        expect(dismissAlert).toHaveBeenCalledWith(uuid);
      });
    });
  });

  describe('when the alert has an icon', () => {
    const alertWithIcon: IAlert = {
      ...alert,
      icon: faSatelliteDish,
    };

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Alert alert={alertWithIcon} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when the alert type is "failure"', () => {
    const failureAlert: IAlert = {
      ...alert,
      type: 'failure',
    };

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Alert alert={failureAlert} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    describe('when the alert has an icon', () => {
      const alertWithIcon: IAlert = {
        ...failureAlert,
        icon: faSatelliteDish,
      };

      it('should match the snapshot', () => {
        const { asFragment } = render(
          <Alert alert={alertWithIcon} />
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('when the alert type is "success"', () => {
    const successAlert: IAlert = {
      ...alert,
      type: 'success',
    };

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Alert alert={successAlert} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    describe('when the alert has an icon', () => {
      const alertWithIcon: IAlert = {
        ...successAlert,
        icon: faSatelliteDish,
      };

      it('should match the snapshot', () => {
        const { asFragment } = render(
          <Alert alert={alertWithIcon} />
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('when the alert type is "warning"', () => {
    const warningAlert: IAlert = {
      ...alert,
      type: 'warning',
    };

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Alert alert={warningAlert} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    describe('when the alert has an icon', () => {
      const alertWithIcon: IAlert = {
        ...warningAlert,
        icon: faSatelliteDish,
      };

      it('should match the snapshot', () => {
        const { asFragment } = render(
          <Alert alert={alertWithIcon} />
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
