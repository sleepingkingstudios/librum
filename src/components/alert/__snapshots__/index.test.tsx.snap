// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-info"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-info pr-2"
      data-icon="circle-info"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;

exports[`<Alert /> when the alert has an icon should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-info"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-satellite-dish pr-2"
      data-icon="satellite-dish"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 32c0-17.7 14.3-32 32-32C383.1 0 512 128.9 512 288c0 17.7-14.3 32-32 32s-32-14.3-32-32C448 164.3 347.7 64 224 64c-17.7 0-32-14.3-32-32zM60.6 220.6L164.7 324.7l28.4-28.4c-.7-2.6-1.1-5.4-1.1-8.3c0-17.7 14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32c-2.9 0-5.6-.4-8.3-1.1l-28.4 28.4L291.4 451.4c14.5 14.5 11.8 38.8-7.3 46.3C260.5 506.9 234.9 512 208 512C93.1 512 0 418.9 0 304c0-26.9 5.1-52.5 14.4-76.1c7.5-19 31.8-21.8 46.3-7.3zM224 96c106 0 192 86 192 192c0 17.7-14.3 32-32 32s-32-14.3-32-32c0-70.7-57.3-128-128-128c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;

exports[`<Alert /> when the alert is dismissable should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-info"
  >
    <button
      class="float-right link-plain"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-x "
        data-icon="x"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z"
          fill="currentColor"
        />
      </svg>
    </button>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-info pr-2"
      data-icon="circle-info"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;

exports[`<Alert /> when the alert type is "failure" should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-failure"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-bomb pr-2"
      data-icon="bomb"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;

exports[`<Alert /> when the alert type is "failure" when the alert has an icon should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-failure"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-satellite-dish pr-2"
      data-icon="satellite-dish"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 32c0-17.7 14.3-32 32-32C383.1 0 512 128.9 512 288c0 17.7-14.3 32-32 32s-32-14.3-32-32C448 164.3 347.7 64 224 64c-17.7 0-32-14.3-32-32zM60.6 220.6L164.7 324.7l28.4-28.4c-.7-2.6-1.1-5.4-1.1-8.3c0-17.7 14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32c-2.9 0-5.6-.4-8.3-1.1l-28.4 28.4L291.4 451.4c14.5 14.5 11.8 38.8-7.3 46.3C260.5 506.9 234.9 512 208 512C93.1 512 0 418.9 0 304c0-26.9 5.1-52.5 14.4-76.1c7.5-19 31.8-21.8 46.3-7.3zM224 96c106 0 192 86 192 192c0 17.7-14.3 32-32 32s-32-14.3-32-32c0-70.7-57.3-128-128-128c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;

exports[`<Alert /> when the alert type is "success" should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-check pr-2"
      data-icon="circle-check"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;

exports[`<Alert /> when the alert type is "success" when the alert has an icon should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-satellite-dish pr-2"
      data-icon="satellite-dish"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 32c0-17.7 14.3-32 32-32C383.1 0 512 128.9 512 288c0 17.7-14.3 32-32 32s-32-14.3-32-32C448 164.3 347.7 64 224 64c-17.7 0-32-14.3-32-32zM60.6 220.6L164.7 324.7l28.4-28.4c-.7-2.6-1.1-5.4-1.1-8.3c0-17.7 14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32c-2.9 0-5.6-.4-8.3-1.1l-28.4 28.4L291.4 451.4c14.5 14.5 11.8 38.8-7.3 46.3C260.5 506.9 234.9 512 208 512C93.1 512 0 418.9 0 304c0-26.9 5.1-52.5 14.4-76.1c7.5-19 31.8-21.8 46.3-7.3zM224 96c106 0 192 86 192 192c0 17.7-14.3 32-32 32s-32-14.3-32-32c0-70.7-57.3-128-128-128c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;

exports[`<Alert /> when the alert type is "warning" should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-warning"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-exclamation pr-2"
      data-icon="circle-exclamation"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24V264c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;

exports[`<Alert /> when the alert type is "warning" when the alert has an icon should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-warning"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-satellite-dish pr-2"
      data-icon="satellite-dish"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 32c0-17.7 14.3-32 32-32C383.1 0 512 128.9 512 288c0 17.7-14.3 32-32 32s-32-14.3-32-32C448 164.3 347.7 64 224 64c-17.7 0-32-14.3-32-32zM60.6 220.6L164.7 324.7l28.4-28.4c-.7-2.6-1.1-5.4-1.1-8.3c0-17.7 14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32c-2.9 0-5.6-.4-8.3-1.1l-28.4 28.4L291.4 451.4c14.5 14.5 11.8 38.8-7.3 46.3C260.5 506.9 234.9 512 208 512C93.1 512 0 418.9 0 304c0-26.9 5.1-52.5 14.4-76.1c7.5-19 31.8-21.8 46.3-7.3zM224 96c106 0 192 86 192 192c0 17.7-14.3 32-32 32s-32-14.3-32-32c0-70.7-57.3-128-128-128c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
        fill="currentColor"
      />
    </svg>
    Attempting to connect to Moon base...
  </div>
</DocumentFragment>
`;
